ydb/core/blobstorage/nodewarden/node_warden_pdisk.cpp

```diff
@@ -281,13 +285,25 @@ namespace NKikimr::NStorage {
     }
 
     void TNodeWarden::OnPDiskRestartFinished(ui32 pdiskId, NKikimrProto::EReplyStatus status) {
-        if (PDiskRestartInFlight.erase(pdiskId) == 0) {
+        auto it = PDiskRestartInFlight.find(pdiskId);
+        if (it == PDiskRestartInFlight.end()) {
             // There was no restart in progress.
             return;
         }
 
+        bool requiresAnotherRestart = it->second;
+
+        PDiskRestartInFlight.erase(it);
```

Comment:
I think you shouldn't erase it if requiresAnotherRestart is true. Because later on, than the restart finishes, lines 289-291 will be executed and no VDisks will be started.